samples/iree-reduce/softmax.mlir

```diff
@@ -0,0 +1,91 @@
+// RUN: iree-compile %s --iree-hal-target-backends=cuda \
+// RUN:     --iree-opt-const-expr-hoisting=false --iree-opt-const-eval=false \
+// RUN:     --iree-codegen-llvmgpu-enable-transform-dialect-jit=false \
+// RUN:     --iree-flow-dispatch-use-transform-dialect=%p/softmax_dispatch_spec.mlir \
+// RUN:     --iree-codegen-llvmgpu-use-transform-dialect=%p/softmax_codegen_spec.mlir | \
+// RUN: iree-run-module --module=- --function=softmax --device=cuda | \
```

Comment:
use VMVX and drop the transform dialect stuff - CUDA is not always available (either built into the compiler or a CUDA device at runtime) and the transform dialect is not required to demonstrate the reducer in use